shared/chat/conversation/messages/wrapper/shared.js

```diff
@@ -12,20 +12,82 @@ const LeftMarker = ({author, isYou, isFollowing, isBroken}) => (
   <Box style={{..._leftMarkerStyle, backgroundColor: marginColor(author, isYou, isFollowing, isBroken)}} />
 )
 
-const UserAvatar = ({author, showImage}) => (
-  <Box style={_userAvatarStyle}>
-    {showImage && <Avatar size={24} username={author} skipBackground={true} />}
-  </Box>
-)
+type UserAvatarProps = {
+  author: string,
+  showImage: boolean,
+  onClick: (author: string) => void,
+}
+
+class UserAvatar extends PureComponent<void, UserAvatarProps, void> {
+  _onClick: () => void
+
+  constructor(props: UserAvatarProps) {
+    super(props)
+
+    this._updateOnClick(props)
+  }
 
-const Username = ({author, isYou, isFollowing, isBroken, includeHeader}) => {
-  if (!includeHeader) return null
-  const style = {
-    color: colorForAuthor(author, isYou, isFollowing, isBroken),
-    ...(isYou ? globalStyles.italic : null),
-    marginBottom: 2,
+  _updateOnClick(p: UserAvatarProps) {
+    const {onClick, author} = p
+    this._onClick = () => {
```

Comment:
we should not be updating these private closures like this in general. if you want to avoid passing a dynamically made closure as a prop its likely better to do

```
// You can assign an arrow function which closes over 'this' so you have access to this.props
_onClick = () => {
  onClick && onClick(this.props.author)
}

...
// also note you have to use ClickableBox else this won't work on native
return (<ClickableBox onClick={this._onClick} />
```